src/sdp_proxy/discovery_proxy.cpp

```diff
@@ -214,11 +214,9 @@ void DiscoveryProxy::OnServiceDiscovered(const std::string
         {
         case OT_DNSSD_QUERY_TYPE_BROWSE:
             splitError = SplitFullServiceName(queryName, serviceName, domain);
-            assert(splitError == OTBR_ERROR_NONE);
             break;
         case OT_DNSSD_QUERY_TYPE_RESOLVE:
             splitError = SplitFullServiceInstanceName(queryName, instanceName, serviceName, domain);
-            assert(splitError == OTBR_ERROR_NONE);
             break;
         default:
             splitError = OTBR_ERROR_NOT_FOUND;
```

Comment:
Please add a comment in the if below (hidden):
```
        if (splitError != OTBR_ERROR_NONE)
        {
            // Incoming service/instance was not what current query wanted to see, move on.
            continue;
        }
```

Reply:
Added in new push.